modules/git/repo_branch_nogogit.go

```diff
@@ -51,32 +51,19 @@ func callShowRef(repoPath, prefix, arg string) ([]string, error) {
 
 	bufReader := bufio.NewReader(stdoutReader)
 	for {
-		// The output of show-ref is simply a list:
-		// <sha> SP <ref> LF
-		_, err := bufReader.ReadSlice(' ')
-		for err == bufio.ErrBufferFull {
-			// This shouldn't happen but we'll tolerate it for the sake of peace
-			_, err = bufReader.ReadSlice(' ')
-		}
-		if err == io.EOF {
-			return branchNames, nil
-		}
-		if err != nil {
-			return nil, err
-		}
-
+		// The output of branch is simply a list:
+		// LF
 		branchName, err := bufReader.ReadString('\n')
 		if err == io.EOF {
-			// This shouldn't happen... but we'll tolerate it for the sake of peace
 			return branchNames, nil
 		}
 		if err != nil {
+			// This shouldn't happen... but we'll tolerate it for the sake of peace
```

Comment:
?

Reply:
I mean the old comment  is wrong, right? The position is where the return value is. So i moved it here which makes more sense to me.

Reply:
Or maybe we should just delete this comment. The old position just does not seem right to me.

Reply:
I think it should still be the original position. It means `err == io.EOF` in fact is not an err. At a normal situation, reader shouldn't return that.

Reply:
@lunny has the correct interpretation. Please revert

Reply:
Am I misunderstanding this? The `err == io.EOF` seems to be the only way to break out the loop normally, but "at a normal situation, reader shouldn't return that"? How should we break out the loop, though